source/MaterialXCore/Document.cpp

```diff
@@ -1258,28 +1331,44 @@ void Document::upgradeVersion()
         }   
 
         // Convert parameters to inputs, applying uniform markings as needed.
-        for (ElementPtr interface : traverseTree())
+        const string FRAME_OFFSET_STRING = "frameoffset";
+        const string INDEX_STRING = "index";
+        const string DEFAULT_STRING = "default";
+        for (ElementPtr elementPtr : traverseTree())
         {
-            if (interface->isA<InterfaceElement>())
+            if (elementPtr->isA<InterfaceElement>())
             {
-                for (ElementPtr param : interface->getChildrenOfType<Element>("parameter"))
+                for (ElementPtr param : elementPtr->getChildrenOfType<Element>("parameter"))
                 {
-                    InputPtr input = updateChildSubclass<Input>(interface, param);
-                    if (interface->isA<NodeDef>())
+                    InputPtr input = updateChildSubclass<Input>(elementPtr, param);
+                    if (elementPtr->isA<NodeDef>())
                     {
-                        // Only strings and filename types should be set as uniforms.
+                        // Strings and filename types should always be uniforms.
                         const string& inputType = input->getType();
                         if (inputType == FILENAME_TYPE_STRING || inputType == STRING_TYPE_STRING)
                         {
                             input->setIsUniform(true);
                         }
+                        // Some integer inputs should be set as uniforms
```

Comment:
Special casing to mark some integer inputs as uniforms (mainly affects image, and geometry inputs)